backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/checkbox/BpkCheckbox.kt

```diff
@@ -0,0 +1,167 @@
+    interactionSource = interactionSource,
+    enabled = enabled,
+    modifier = modifier,
+    content = { BpkText(text) },
+  )
+}
+
+@Composable
+fun BpkCheckbox(
+  text: String,
+  state: ToggleableState,
+  onClick: (() -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+) {
+  BpkCheckbox(
+    state = state,
+    onClick = onClick,
+    modifier = modifier,
+    enabled = enabled,
+    interactionSource = interactionSource,
+    content = { BpkText(text) },
+  )
+}
+
+@Composable
+fun BpkCheckbox(
+  checked: Boolean,
+  onCheckedChange: ((Boolean) -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+  content: @Composable RowScope.(Boolean) -> Unit,
+) {
+  BpkCheckbox(
+    state = ToggleableState(checked),
+    onClick = if (onCheckedChange != null) { { onCheckedChange(!checked) } } else null,
+    interactionSource = interactionSource,
+    enabled = enabled,
+    modifier = modifier,
+    content = { content(checked) },
+  )
+}
+
+@Composable
+fun BpkCheckbox(
+  state: ToggleableState,
+  onClick: (() -> Unit)?,
+  modifier: Modifier = Modifier,
+  enabled: Boolean = true,
+  interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+  content: @Composable RowScope.(ToggleableState) -> Unit,
+) {
+  Row(
+    verticalAlignment = Alignment.CenterVertically,
+    horizontalArrangement = Arrangement.SpaceBetween,
+    modifier = modifier.applyIf(onClick != null) {
+      triStateToggleable(
+        state = state,
+        interactionSource = interactionSource,
+        indication = null,
+        enabled = enabled,
+        role = Role.Checkbox,
+        onClick = onClick!!,
+      )
+    },
+  ) {
+
+    BpkCheckboxImpl(
+      modifier = Modifier.padding(end = BpkSpacing.Sm),
+      state = state,
+      enabled = enabled,
+      interactionSource = interactionSource,
+      onClick = onClick ?: {},
+    )
+
+    BpkToggleableContent(
+      enabled = enabled,
+      content = { content(state) }
+    )
+
+  }
+}
+
+@OptIn(ExperimentalMaterialApi::class, androidx.compose.ui.ExperimentalComposeUiApi::class)
+@Composable
+private fun BpkCheckboxImpl(
+  state: ToggleableState,
+  onClick: () -> Unit,
+  modifier: Modifier = Modifier,
+  enabled: Boolean,
+  interactionSource : MutableInteractionSource,
+) {
+  // our design system isn't designed with the minimum touch target in mind at the moment.
+  // Disable the enforcement to avoid the extra padding
+  CompositionLocalProvider(LocalMinimumTouchTargetEnforcement provides false) {
+    TriStateCheckbox(
+      state = state,
+      onClick = onClick,
+      enabled = enabled,
+      modifier = modifier.scale(0.89f).semantics { invisibleToUser() },
+      interactionSource = interactionSource,
+      colors = CheckboxDefaults.colors(
```

Comment:
looks like we're setting a lot more colours in the view version - are those set correctly by the theme?